src/util/test/pagination-test.js

```diff
@@ -1,22 +1,115 @@
-import { pageNumberOptions } from '../pagination';
-
-describe('sidebar/util/pagination', () => {
-  describe('pageNumberOptions', () => {
-    [
-      { args: [1, 10, 5], expected: [1, 2, 3, 4, null, 10] },
-      { args: [3, 10, 5], expected: [1, 2, 3, 4, null, 10] },
-      { args: [6, 10, 5], expected: [1, null, 5, 6, 7, null, 10] },
-      { args: [9, 10, 5], expected: [1, null, 7, 8, 9, 10] },
-      { args: [2, 3, 5], expected: [1, 2, 3] },
-      { args: [3, 10, 7], expected: [1, 2, 3, 4, 5, 6, null, 10] },
-      { args: [1, 1, 5], expected: [] },
-    ].forEach(testCase => {
-      it('should produce expected available page numbers', () => {
-        assert.deepEqual(
-          pageNumberOptions(...testCase.args),
-          testCase.expected,
-        );
+import { paginationItems } from '../pagination';
+
+describe('paginationItems', () => {
+  [
+    // Only one page
+    {
+      current: 1,
+      total: 1,
+      expected: [],
+    },
+
+    // No pages elided.
+    {
+      current: 1,
+      total: 3,
+      expected: [1, 2, 3],
+    },
+
+    // Pages elided after current.
+    {
+      current: 1,
+      total: 4,
+      expected: [1, 2, null, 4],
+    },
+
+    // Pages elided before current.
+    {
+      current: 4,
+      total: 6,
+      expected: [1, null, 3, 4, 5, 6],
+    },
+
+    // Pages elided before and after current.
+    {
+      current: 4,
+      total: 7,
+      expected: [1, null, 3, 4, 5, null, 7],
+    },
+
+    // Custom boundary count
+    {
+      current: 6,
+      total: 10,
+      boundaryCount: 2,
+      expected: [1, 2, null, 5, 6, 7, null, 9, 10],
+    },
+
+    // Custom sibling count
+    {
+      current: 6,
+      total: 10,
+      siblingCount: 2,
+      expected: [1, null, 4, 5, 6, 7, 8, null, 10],
+    },
+  ].forEach(({ current, total, boundaryCount, siblingCount, expected }) => {
+    it('should produce expected items', () => {
+      const items = paginationItems(current, total, {
+        boundaryCount,
+        siblingCount,
       });
+      assert.deepEqual(items, expected);
     });
   });
+
+  function* configurations() {
```

Comment:
The test above checks a few specific examples against expected results. The test below is a property test that generates a range of configurations and for each verifies that certain properties are satisfied for every possible `current` page.